### PR TITLE
fix esm2 downstream task finetune config saving issue

### PR DESCRIPTION
**Pull request by zmsunnyday**
### Description
When doing esm2 downstream task finetuning, the task head related configs (mlp,cnn) are not saved correctly in the checkpoint files, causing tensor shape mismatching error when doing inference. This PR is to fix this issue.

### Type of changes

- [x]  Bug fix (non-breaking change which fixes an issue)
- [ ]  New feature (non-breaking change which adds functionality)
- [ ]  Refactor
- [ ]  Documentation update
- [ ]  Other (please describe):

### CI Pipeline Configuration
Configure CI behavior by applying the relevant labels:

- [SKIP_CI](https://github.com/NVIDIA/bionemo-framework/blob/main/docs/docs/user-guide/contributing/contributing.md#skip_ci) - Skip all continuous integration tests
- [INCLUDE_NOTEBOOKS_TESTS](https://github.com/NVIDIA/bionemo-framework/blob/main/docs/docs/user-guide/contributing/contributing.md#include_notebooks_tests) - Execute notebook validation tests in pytest
- [INCLUDE_SLOW_TESTS](https://github.com/NVIDIA/bionemo-framework/blob/main/docs/docs/user-guide/contributing/contributing.md#include_slow_tests) - Execute tests labelled as slow in pytest for extensive testing

> [!NOTE]
> By default, the notebooks validation tests are skipped unless explicitly enabled.

#### Authorizing CI Runs

We use [copy-pr-bot](https://docs.gha-runners.nvidia.com/apps/copy-pr-bot/#automation) to manage authorization of CI
runs on NVIDIA's compute resources.

* If a pull request is opened by a trusted user and contains only trusted changes, the pull request's code will
  automatically be copied to a pull-request/ prefixed branch in the source repository (e.g. pull-request/123)
* If a pull request is opened by an untrusted user or contains untrusted changes, an NVIDIA org member must leave an
  `/ok to test` comment on the pull request to trigger CI. This will need to be done for each new commit.

### Usage

```python
TODO: Add code snippet
```

### Pre-submit Checklist

 - [x] I have tested these changes locally
 - [ ] I have updated the documentation accordingly
 - [ ] I have added/updated tests as needed
 - [ ] All existing tests pass successfully

### CODE REVIEW

**Comment by pstjohn** [comment]
Could you add a test? Ideally for a bug fix you'd have a test that fails with the previous version and passes with your fix

**Comment by zmsunnyday** [comment]
Hi @pstjohn , I have attach the sample dataset for testing. Here are the steps to reproduce the issue:
1. finetune the sequence level classification task 
`finetune_esm2 \
    --restore-from-checkpoint-path {pretrain_checkpoint_path} \
    --train-data-path {data_path} \
    --valid-data-path {data_path} \
    --config-class ESM2FineTuneSeqConfig \
    --dataset-class InMemorySingleValueDataset \
    --task-type "classification" \
    --mlp-ft-dropout 0.25 \
    --mlp-hidden-size 256 \
    --mlp-target-size 3 \
    --experiment-name "sequence-level-classification" \
    --num-steps 50 \
    --num-gpus 1 \
    --val-check-interval 10 \
    --log-every-n-steps 10 \
    --encoder-frozen \
    --lr 5e-3 \
    --lr-multiplier 1e2 \
    --scale-lr-layer "classification_head" \
    --result-dir {work_dir}  \
    --micro-batch-size 2 \
    --num-gpus 1 \
    --precision "bf16-mixed"`
2.do inference
`infer_esm2 --checkpoint-path {checkpoint_path} \
             --config-class ESM2FineTuneTokenConfig \
             --data-path {data_path} \
             --results-path {results_path} \
             --micro-batch-size 3 \
             --num-gpus 1 \
             --precision "bf16-mixed" \
             --include-embeddings \
             --include-hiddens \
             --include-input-ids`
The you will get an error `Global shape mismatch for loaded (torch.Size([3, 256])) and expected ((1, 256)) tensor for key module.classification_head.linear_layers.1.weight`
After updating the finetune_esm2.py with my changes, the bug will be fixed.
[classification_data.csv](https://github.com/user-attachments/files/19881778/classification_data.csv)
